### PR TITLE
First time flow updates

### LINKED ISSUE #6163: Streamline onboarding flow
We can simplify the onboarding flow with the following:
* add ToS checkbox + link to password screens
* remove "privacy warning" as it's explained during 1102 / enable flow
* add security warning to Jazzicon intro step
* introduce import vs. create step 

See flow [here](https://www.figma.com/file/EQ9FaiXKyXx9uHt62RO8kv5A/onboarding)

### PR DESCRIPTION
**Pull request by danjm**
fixes #6163 

Final design decisions pending. Tests will be updated once design decisions finalized.

peek 2019-02-20 16-52

----

peek 2019-02-20 13-59

----
peek 2019-02-20 21-40

### CODE REVIEW

File: ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js [hunk 1]
```diff
@@ -131,20 +131,26 @@ export default class ImportWithSeedPhrase extends PureComponent {
     return !passwordError && !confirmPasswordError && !seedPhraseError
   }
 
+  toggleTermsCheck = () => {
+    this.setState({
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js]: We should prefer [the `updater` function argument](https://reactjs.org/docs/react-component.html#setstate) here. That is,

```js
this.setState((prevState) => ({
    termsChecked: !prevState.termsChecked,
})
```

File: ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js [hunk 2]
```diff
@@ -197,19 +203,22 @@ export default class ImportWithSeedPhrase extends PureComponent {
           margin="normal"
           largeLabel
         />
+        <div className="first-time-flow__checkbox-container" onClick={this.toggleTermsCheck}>
+          <div className="first-time-flow__checkbox">
+            {termsChecked ? <i className="fa fa-check fa-2x" /> : null}
+          </div>
+          <span className="first-time-flow__checkbox-label">
+            I agree to the Terms Of Service
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js]: Should we extract this into a message string?

File: ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js [hunk 3]
```diff
@@ -111,12 +112,29 @@ export default class NewAccount extends PureComponent {
     history.push(INITIALIZE_IMPORT_WITH_SEED_PHRASE_ROUTE)
   }
 
+  toggleTermsCheck = () => {
+    this.setState({
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js]: We should prefer [the `updater` function argument](https://reactjs.org/docs/react-component.html#setstate) here. That is,

```js
this.setState((prevState) => ({
    termsChecked: !prevState.termsChecked,
})
```

File: ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js [hunk 4]
```diff
@@ -111,12 +112,29 @@ export default class NewAccount extends PureComponent {
...
       <div>
+        <div className="first-time-flow__create-back">
+          <a
+            onClick={e => {
+              e.preventDefault()
+              this.props.history.push(INITIALIZE_SELECT_ACTION_ROUTE)
+            }}
+            href="#"
+          >
+            {`< Back`}
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js]: Should we extract this into a message string?

File: ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js [hunk 5]
```diff
@@ -151,27 +169,23 @@ export default class NewAccount extends PureComponent {
             fullWidth
             largeLabel
           />
+          <div className="first-time-flow__checkbox-container" onClick={this.toggleTermsCheck}>
+            <div className="first-time-flow__checkbox">
+              {termsChecked ? <i className="fa fa-check fa-2x" /> : null}
+            </div>
+            <span className="first-time-flow__checkbox-label">
+              I agree to the Terms Of Service
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js]: Should we extract this into a message string?

File: ui/app/components/pages/first-time-flow/create-password/unique-image/unique-image.component.js [hunk 6]
```diff
@@ -1,52 +1,51 @@
...
         <div className="first-time-flow__text-block">
-          { t('yourUniqueAccountImageDescription2') }
+          { t('protectYourKeysMessage2') }
         </div>
         <Button
-          type="first-time"
+          type="confirm"
           className="first-time-flow__button"
-          onClick={() => history.push(INITIALIZE_NOTICE_ROUTE)}
+          onClick={async () => {
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/create-password/unique-image/unique-image.component.js]: Why is this handler `async`?

File: ui/app/components/pages/first-time-flow/end-of-flow/end-of-flow.component.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        </div>
+        <Button
+          type="confirm"
+          className="first-time-flow__button"
+          onClick={async () => {
+            await completeOnboarding()
+            history.push(DEFAULT_ROUTE)
+          }}
+        >
+          { 'All Done' }
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/end-of-flow/end-of-flow.component.js]: Can we inline all of the strings in this component? Or maybe we should even extract them into message strings?

File: ui/app/components/pages/first-time-flow/select-action/select-action.component.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Button from '../../../button'
+import { INITIALIZE_CREATE_PASSWORD_ROUTE, INITIALIZE_NOTICE_ROUTE, INITIALIZE_IMPORT_WITH_SEED_PHRASE_ROUTE } from '../../../../routes'
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/select-action/select-action.component.js]: Nitpick: can we split this import onto multiple lines?

File: ui/app/components/pages/first-time-flow/select-action/select-action.component.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+              </div>
+              <div className="select-action__select-button">
+                <div className="select-action__button-symbol">
+                  <img src="/images/download-alt.svg" />
+                </div>
+                <div className="select-action__button-text-big">
+                  { 'Import Wallet' }
+                </div>
+                <div className="select-action__button-text-small">
+                  { 'Import your existing wallet using a 12 word seed phrase' }
```
Comment (whymarrh) [inline ui/app/components/pages/first-time-flow/select-action/select-action.component.js]: Should we extract the messages in this component into message strings?

**Review by whymarrh — CHANGES_REQUESTED** [review]
I left a few comments inline. Also, if we're removing the notices should we also remove the components and files (the `notices/` directory?) associated with them?

File: app/_locales/en/messages.json [hunk 10]
```diff
@@ -848,6 +869,18 @@
...
+    "message": "New to MetaMask?"
+  },
+  "protectYourKeys": {
+    "message": "Protect Your Keys!"
+  },
+  "protectYourKeysMessage1": {
+    "message": "Be careful with your seed phrase — there have been reports of websites that attempt to imitate MetaMask. MetaMask will never spontaneously ask for your seed phrase!"
+  },
+  "protectYourKeysMessage2": {
+    "message": "Keep your phrase safe. If you see something phishy, or you’re uncertain about a website, email support@metamask.io."
```
[the remark this answers is not in the document]
Comment (danjm) [reply app/_locales/en/messages.json]: Done in e7b8ea021

File: test/e2e/beta/run-all.sh [hunk 11]
```diff
@@ -6,6 +6,6 @@ set -o pipefail
 
 export PATH="$PATH:./node_modules/.bin"
 
-shell-parallel -s 'npm run ganache:start -- -b 2' -x 'sleep 5 && static-server test/e2e/beta/contract-test --port 8080' -x 'sleep 5 && mocha test/e2e/beta/metamask-beta-ui.spec'
+# shell-parallel -s 'npm run ganache:start -- -b 2' -x 'sleep 5 && static-server test/e2e/beta/contract-test --port 8080' -x 'sleep 5 && mocha test/e2e/beta/metamask-beta-ui.spec'
```
[the remark this answers is not in the document]
Comment (danjm) [reply test/e2e/beta/run-all.sh]: Fixed in ac872717b

File: ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js (hunk 1 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js]: Done in ac872717b

File: ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js (hunk 2 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/create-password/import-with-seed-phrase/import-with-seed-phrase.component.js]: Done in e7b8ea021

File: ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js (hunk 3 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js]: Fixed in 51fb72b40

File: ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js (hunk 5 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/create-password/new-account/new-account.component.js]: Done in e7b8ea021

File: ui/app/components/pages/first-time-flow/create-password/unique-image/unique-image.component.js (hunk 6 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/create-password/unique-image/unique-image.component.js]: Fixed in ac872717b

File: ui/app/components/pages/first-time-flow/end-of-flow/end-of-flow.component.js (hunk 7 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/end-of-flow/end-of-flow.component.js]: Fixed in e7b8ea021

File: ui/app/components/pages/first-time-flow/select-action/select-action.component.js (hunk 8 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/select-action/select-action.component.js]: Fixed in ac872717b

File: ui/app/components/pages/first-time-flow/select-action/select-action.component.js (hunk 9 above)
Comment (danjm) [reply ui/app/components/pages/first-time-flow/select-action/select-action.component.js]: Done in e7b8ea021

**Comment by danjm** [comment]
@whymarrh noticed related code removed in 275f7c247

File: app/_locales/en/messages.json [hunk 12]
```diff
@@ -448,6 +463,24 @@
   "encryptNewDen": {
     "message": "Encrypt your new DEN"
   },
+  "endOfFlowMessage1": {
+    "message": "You passed the test - keep your seedphrase safe, it's your responsibility!"
+  },
+  "endOfFlowMessage2": {
+    "message": "Tips on storing it safely "
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nitpick: trailing space

File: app/_locales/en/messages.json [hunk 13]
```diff
@@ -448,6 +463,24 @@
...
+  },
+  "endOfFlowMessage3": {
+    "message": "• Save a backup in multiple places"
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nit: can we factor out the bullet?

File: app/_locales/en/messages.json [hunk 14]
```diff
@@ -448,6 +463,24 @@
...
+  },
+  "endOfFlowMessage4": {
+    "message": "• Never tell anyone"
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nit: can we factor out the bullet?

File: app/_locales/en/messages.json [hunk 15]
```diff
@@ -448,6 +463,24 @@
...
+  },
+  "endOfFlowMessage5": {
+    "message": "If you need to back it up again, you can find them in Settings -> Security."
+  },
+  "endOfFlowMessage6": {
+    "message": "*Metamask cannot recover your seedphrase. Learn more."
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nit: can we factor out the bullet?

Comment (whymarrh) [reply app/_locales/en/messages.json]: Nit: also `s/Metamask/MetaMask/`

File: app/_locales/en/messages.json [hunk 16]
```diff
@@ -448,6 +463,24 @@
...
+    "message": "Tips on storing it safely "
+  },
+  "endOfFlowMessage3": {
+    "message": "• Save a backup in multiple places"
+  },
+  "endOfFlowMessage4": {
+    "message": "• Never tell anyone"
+  },
+  "endOfFlowMessage5": {
+    "message": "If you need to back it up again, you can find them in Settings -> Security."
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nit: `s/find them/find it/` I think? Or maybe `s/back it up/back them up/` if that works?

File: app/_locales/en/messages.json [hunk 17]
```diff
@@ -848,6 +893,21 @@
...
+    "message": "New to MetaMask?"
+  },
+  "noAlreadyHaveSeed": {
+    "message": "No, I already have a seed phrase"
+  },
+  "protectYourKeys": {
+    "message": "Protect Your Keys!"
+  },
+  "protectYourKeysMessage1": {
+    "message": "Be careful with your seed phrase — there have been reports of websites that attempt to imitate MetaMask. MetaMask will never spontaneously ask for your seed phrase!"
```
Comment (whymarrh) [inline app/_locales/en/messages.json]: Nit: should we drop "spontaneously"? Below we have `yourUniqueAccountImageDescription3` plainly saying never.

**Review by whymarrh — CHANGES_REQUESTED** [review]

**Review by whymarrh — APPROVED** [review]

**Pull request merged**